app/mailers/candidate_mailer.rb

```diff
@@ -467,12 +467,12 @@ def fraud_match_email(application_form)
 
   def new_offer(application_choice, template_name)
     @application_choice = application_choice
-    course_option = CourseOption.find_by(id: @application_choice.current_course_option_id) || @application_choice.course_option
+    course_option = CourseOption.find_by(id: @application_choice.current_course_option_id) || @application_choice.current_course_option
     @provider_name = course_option.course.provider.name
     @course_name = course_option.course.name_and_code
     @conditions = @application_choice.offer.conditions_text
     @offers = @application_choice.self_and_siblings.select(&:offer?).map do |offer|
-      "#{offer.course_option.course.name_and_code} at #{offer.course_option.course.provider.name}"
+      "#{offer.current_course_option.course.name_and_code} at #{offer.current_course_option.course.provider.name}"
```

Comment:
I think offer just has `course_option`?

Reply:
it does, have fixed, thank you!

Reply:
Is `offer` a badly named variable? It looks like the `map` is iterating over application choices not offers? In which case `current_course_option` was right.

Reply:
Thanks, i've renamed the variable inside the block to try and make it easier to understand but i've kept it as `course_option` because of the final line in your comment